src/convertToObject.js

```diff
@@ -6,7 +6,23 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const styles = sourceString
+    .trim()
+    .split(';')
+    .filter((style) => style.trim() !== '');
+
+  const stylesObject = styles.reduce((acc, style) => {
+    const [property, value] = style.split(':', 2)
+      .map((item) => item.trim());
+
+    if (property.length > 0 && value.length > 0) {
+      acc[property] = value;
+    }
+
+    return acc;
```

Comment:
The `reduce` method is used correctly to transform the array into an object. However, you should handle the case where the `style` string does not contain a colon, which would result in `value` being `undefined`. This would cause an issue when trying to call `trim` on it. Make sure to check that both `property` and `value` are defined before trimming and adding them to the accumulator.